src/zope/file/testing.py

```diff
@@ -146,3 +136,31 @@ def _extractContentInfo(self, item):
         info['object'] = obj
         info['url'] = urllib.quote(oid.encode('utf-8'))
         return info
+
+@implementer(IBrowserPublisher)
+class ManagementViewSelector(BrowserView):
+    """View that selects the first available management view.
+
+    Support 'zmi_views' actions like: 'javascript:alert("hello")',
+    '../view_on_parent.html' or '++rollover++'.
+    """
+    # Copied from zope.app.publication
+
+    def browserDefault(self, request):
+        return self, ()
+
+    def __call__(self):
+        item = getFirstMenuItem('zmi_views', self.context, self.request)
+
+        if item:
+            redirect_url = item['action']
+            if not (redirect_url.startswith('../') or \
+                    redirect_url.lower().startswith('javascript:') or \
+                    redirect_url.lower().startswith('++')):
```

Comment:
You don't need the `\` inside `(`...`)`.

You don't need to `.lower()` to check for a `++` prefix.

You could merge multiple `.startswith()` checks into one `.startswith(('../', 'javascript:', '++'))`.